### ISSUE TITLE
Porting to MahApps 2.x

### ISSUE THREAD

**Issue by RickStrahl** [RickStrahl/MarkdownMonster#672]
I'm in the process of updating Markdown Monster to MahApps 2.x (2.1 currently). I've updated packages and fixed the many compile time changes, so MM at least runs at this point and is functional. But there are major styling issues - mainly due to any of the old custom stylings that were previously applied no longer working due to name and namespace changes and general changes in how styles are rendered in 2.x.

I'm looking for some help with getting the styles whipped back into place. Here are some specific areas that are currently broken and need to be addressed:

* ~~Toolbar Menu bar - sizing, missing icons for split button it looks like~~
   * ~~Looks like padding and fontsize are the problem. Font-size is not bubbling down from toolbar into items.  Padding at base style is too large~~

* ~~ComboBox backgrounds don't respect transparency~~
* ~~Menus and drop downs have white border - should be soft  dark border~~
* ~~Textboxes are generally sized wrong, with incorrect padding, sizing and alignment~~
* ~~Splitters are not interactively resizing - and the split bars are obscured behind the WebBrowser windows~~
* ~~Light Theme - using `Light.Steel`. Need adjustments for window bar~~

There are likely to be more and I'll keep updating this document as items get fixed and added.

My WPF skills are... not great :smile: and I while I can slowly pound through this, somebody with better WPF foo could probably help speed this process up significantly. So if anybody wants to take a crack at any one of these or help me work through some of these that would be awesome. If you do decide to help, please if possible try to **add small PRs with specific style fixes** rather than one large patch. Mahalo! 

Current work on the MahApps update is in the **MahApps20 Branch** in Git if anybody is interested.

**Comment by punker76** [comment]
@RickStrahl I'll take a look 😉

**Comment by RickStrahl** [comment]
Awesome Jan. I've fixed a few things - basically removing most of the custom stylings brings back a lot of base behavior, so I think that's the way to start. All the color customizations I probably throw out and start over.

**Comment by RickStrahl** [comment]
This went pretty good and MahApps 2.x certainly makes a lot of things a lot easier than they used to be!

Here are a few notes (for my own record):

* Spacing of buttons is very different so almost all toolbars and button groups needed adjusting (basically removing margins that were previously required. 

* The Dark Theme's white border is a bit harsh so changing those (for default in MahApps) seems necessary.  `<SolidColorBrush x:Key="MahApps.Brushes.SubMenu.Border" Color="#606060" /` and `<SolidColorBrush x:Key="MahApps.Brushes.ContextMenu.Border" Color="#606060" />` handle these now. Nice!

* Splitters now use a Preview rather than live resizing. WOuld love to use that but doesn't work for MM due to the airspace issues with the Web Browser control. Requres using `ShowPreview=false;` to keep the old, live-resizing behavior.

* Light Themes have changed styling a bit. Specifically there's only one theme that doesn't have a white foreground, which mucked with my customized themes. I ended up having to explicitly override the theme foreground color in ` <SolidColorBrush x:Key="MahApps.Brushes.IdealForeground" Color="#222" />` for the light theme to affect title and status bar font colors.

Overall not a ton of changes, although not obvious. I was able to remove a lot of the old style overrides that Mah now handles much better by default. What's left was refactored down to a very small number of style overrides which is awesome for future maintenance.
